vividus-extension-selenium/src/main/java/org/vividus/proxy/mitm/MitmManagerFactory.java

```diff
@@ -37,6 +41,21 @@ public MitmManager createMitmManager(MitmManagerOptions options)
         checkNotNull(keyStoreOptions.getPassword(), "key store password");
         checkNotNull(options.getAlias(), "alias");
 
+        MitmManager mitmManager;
+        if (options.getMitmManagerType() == MitmManagerType.IMPERSONATED)
+        {
+            mitmManager = buildImpersonatingMitmManager(options, keyStoreOptions);
+        }
+        else
+        {
+            mitmManager = buildSelfSignedMitmManager(options, keyStoreOptions);
+        }
```

Comment:
move this logic to MitmManagerType, for example:

`MitmManagerType.IMPERSONATED.buildMitmManager(MitmManagerOptions options, KeyStoreOptions keyStoreOptions)`